api/test/e2e/route_with_vars_test.go

```diff
@@ -0,0 +1,307 @@
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc:     "delete route",
+			Object:       MangerApiExpect(t),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/routes/r1",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc: "create route with multiple vars (args, cookie and header)",
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+					"uri": "/hello",
+					"vars": [
+						["http_cookie","==","_octo=GH1.1.572248189.1598928545; _device_id=2c1a1a52074e66a3a008e4b73c690500; logged_in=yes;"],
+						["http_k","==","header"],
+						["arg_name","==","aaa"]
+					],
+					"upstream": {
+						"type": "roundrobin",
+						"nodes": [{
+							"host": "172.16.238.20",
+							"port": 1980,
+							"weight": 1
+						}]
+					}
+				}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+
+		{
+			caseDesc:     "hit the route with right parameters",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"k": "header", "Cookie": "_octo=GH1.1.572248189.1598928545; _device_id=2c1a1a52074e66a3a008e4b73c690500; logged_in=yes;"},
+			Path:         `/hello`,
+			Query:        "name=aaa",
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc:     "hit the route with wrong arg",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"k": "header", "Cookie": "_octo=GH1.1.572248189.1598928545; _device_id=2c1a1a52074e66a3a008e4b73c690500; logged_in=yes;"},
+			Path:         `/hello`,
+			ExpectStatus: http.StatusNotFound,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc:     "hit the route with wrong header",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"k": "test", "Cookie": "_octo=GH1.1.572248189.1598928545; _device_id=2c1a1a52074e66a3a008e4b73c690500; logged_in=yes;"},
+			Path:         `/hello`,
+			Query:        "name=aaa",
+			ExpectStatus: http.StatusNotFound,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc:     "hit the route with wrong cookie",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"k": "header", "Cookie": "_octo=GH1.1.572248189.1598928545; logged_in=yes;"},
+			Path:         `/hello`,
+			Query:        "name=aaa",
+			ExpectStatus: http.StatusNotFound,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc:     "delete route",
+			Object:       MangerApiExpect(t),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/routes/r1",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
+
+		{
+			caseDesc: "add route with vars (args is digital)",
```

Comment:
again

Reply:
because the old route was deleted at previous test case (to avoid affecting this case), so I think here should be add again